ccaas_builder/cmd/detect/main.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+var logger = log.New(os.Stderr, "", 0)
+
+func main() {
+	logger.Println("::Detect")
+
+	if err := run(); err != nil {
+		logger.Printf("::Error: %v\n", err)
+		os.Exit(1)
+	} else {
+		logger.Printf("::Type detected as ccaas")
+	}
+}
+
+type ChaincodeMetadata struct {
+	Type string `json:"type"`
+}
+
+func run() error {
+	if len(os.Args) > 2 {
+		chaincodeMetaData := os.Args[2]
+		mdbytes, err := ioutil.ReadFile(filepath.Join(chaincodeMetaData, "metadata.json"))
+		if err != nil {
+			return err
+		}
+
+		var metadata ChaincodeMetadata
+		err = json.Unmarshal(mdbytes, &metadata)
+		if err != nil {
+			return err
+		}
+
+		if strings.ToLower(metadata.Type) == "ccaas" {
+			return nil
+		} else {
+			return fmt.Errorf("chaincode type not supported: %s", metadata.Type)
+		}
+
+	} else {
```

Comment:
It is idiomatic to avoid indenting and else statements. Handle the failure first in a short indent, and then the normal flow can be un-indented.
See guidance at https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow